integration-test/background/click-to-load-facebook.js

```diff
@@ -26,61 +70,74 @@ describe('Test Facebook Click To Load', () => {
+        const buttonCount = await page.evaluate(() => {
+            window.buttons =
+                Array.from(document.querySelectorAll('body > div'))
+                    .map(div => div.shadowRoot && div.shadowRoot.querySelector('button'))
+                    .filter(button => button)
+            return window.buttons.length
         })
-
-        expect(Number(fbRequestData.requests)).toBeLessThanOrEqual(Number(fbRequestData.expectedCalls))
-
-        try {
-            await page.close()
-        } catch (e) {}
-    })
-
-    xit('CTL: Should load Facebook elements on click', async () => {
-        const page = await browser.newPage()
-
-        try {
-            await page.goto(testSite, { waitUntil: 'networkidle2', timeout: 10000 })
-        } catch (e) {
-            // timed out waiting for page to load, let's try running the test anyway
-            console.log(`Timed out: ${e}`)
+        for (let i = 0; i < buttonCount; i++) {
+            const button = await page.evaluateHandle(i => window.buttons[i], i)
+            await button.click()
+        }
+        await page.waitForNetworkIdle({ idleTime: 1000 })
+        {
+            const {
+                requestCount, blockCount, allowCount, facebookSDKRedirect
+            } = summariseFacebookRequests(pageRequests)
+
+            expect(facebookSDKRedirect.checked).toBeTrue()
+            expect(facebookSDKRedirect.alwaysRedirected).toBeFalse()
+            expect(requestCount).toBeGreaterThan(5)
+            expect(blockCount).toEqual(0)
+            expect(allowCount).toEqual(requestCount)
         }
-        // give it little time just to be sure (facebook widgets can take time to load)
-        await page.waitForTimeout(4000)
-
-        const fbRequestDataBeforeClick = await page.evaluate(() => {
-            return {
-                requests: document.getElementById('facebook_call_count').innerHTML
-            }
-        })
-
-        // click image element to trigger click to load
-        page.click('div > div > div > button')
-
-        await page.waitForTimeout(6000) // FB elements can take a while to load...
 
-        const fbRequestDataAfterClick = await page.evaluate(() => {
-            return {
-                requests: document.getElementById('facebook_call_count').innerHTML
-            }
-        })
+        // When the page is reloaded, requests should be blocked again.
+        pageRequests.length = 0
+        await page.reload({ waitUntil: 'networkidle0' })
+        await page.waitForNetworkIdle({ idleTime: 1000 })
+        {
+            const {
+                requestCount, blockCount, allowCount, facebookSDKRedirect
+            } = summariseFacebookRequests(pageRequests)
+
+            expect(facebookSDKRedirect.checked).toBeTrue()
+            // FIXME - It seems that requests to the SDK are not reliably
+            //         redirected after the page is reloaded.
+            // expect(facebookSDKRedirect.alwaysRedirected).toBeTrue()
```

Comment:
so weird, we don't see that in the regular browser, right?

Reply:
I suspect that we would, I spent quite a while debugging the tests to make sure it wasn't just a bug in the tests themselves. I filed an Asana task to follow up.

Reply:
OK, fair enough, we can figure this out in the follow up.